cmd/tailscale/cli/serve_dev.go

```diff
@@ -203,85 +203,97 @@ func (e *serveEnv) runServeCombined(subcmd serveMode) execFunc {
 			return errHelp
 		}
 
-		if turnOff {
-			err := e.unsetServe(ctx, srvType, srvPort, mount)
+		sc, err := e.lc.GetServeConfig(ctx)
+		if err != nil {
+			return fmt.Errorf("error getting serve config: %w", err)
+		}
+
+		// nil if no config
+		if sc == nil {
+			sc = new(ipn.ServeConfig)
+		}
+		dnsName := strings.TrimSuffix(st.Self.DNSName, ".")
```

Comment:
nit: could move assignment closer to its usage starting on line 263

Reply:
Sounds good, let's do that in the follow up ✌🏼